compiler/hash-ast/src/ast.rs

```diff
@@ -912,33 +857,53 @@ pub enum ExpressionKind<'c> {
     Typed(TypedExpr<'c>),
     Block(BlockExpr<'c>),
     Import(ImportExpr<'c>),
+    StructDef(StructDef<'c>),
+    EnumDef(EnumDef<'c>),
+    Bound(Bound<'c>),
+    TraitDef(TraitDef<'c>),
+    Return(ReturnStatement<'c>),
+    Break(BreakStatement),
+    Continue(ContinueStatement),
+    Assign(AssignExpression<'c>),
 }
 
 /// An expression.
 #[derive(Debug, PartialEq)]
 pub struct Expression<'c> {
+    /// The kind of the expression
     kind: ExpressionKind<'c>,
 }
 
 impl<'c> Expression<'c> {
+    /// Create a new [Expression] with a specific [ExpressionKind] and with
+    /// no bound
     pub fn new(kind: ExpressionKind<'c>) -> Self {
         Self { kind }
     }
 
+    /// Convert the [Expression] into an [ExpressionKind]
     pub fn into_kind(self) -> ExpressionKind<'c> {
         self.kind
     }
 
+    /// Get the [ExpressionKind] of the expression
     pub fn kind(&self) -> &ExpressionKind<'c> {
         &self.kind
     }
+
+    pub fn no_continuation(&self) -> bool {
+        matches!(
+            self.kind(),
+            ExpressionKind::Import(_) | ExpressionKind::Break(_) | ExpressionKind::Continue(_)
+        )
+    }
```

Comment:
Is this really necessary?

Reply:
If so, maybe a comment as to why would be useful?

Reply:
Yeah I think this was a remnant of the old syntax, I will now remove it.